ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/commands/tracker/CompletionResponse.scala

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.client.services.commands.tracker
+
+import com.daml.ledger.api.v1.completion.Completion
+import com.google.protobuf.Any
+import com.google.rpc
+import com.google.rpc.status.{Status => StatusProto}
+import io.grpc.Status.Code
+import io.grpc.{StatusRuntimeException, protobuf}
+
+import scala.jdk.CollectionConverters._
+
+object CompletionResponse {
+  type CompletionResponse = Either[CompletionFailure, CompletionSuccess]
+  sealed trait CompletionFailure {
+    def metadata: Map[String, String] = Map.empty
+  }
+  final case class NotOkResponse(commandId: String, grpcStatus: StatusProto)
+      extends CompletionFailure
+  final case class TimeoutResponse(commandId: String) extends CompletionFailure
+  final case class NoStatusInResponse(commandId: String) extends CompletionFailure
```

Comment:
In theory it should never happen.
But we have to handle this case because of the type we receive, as the GRPC status is optional in the protobuf `Completion`.